### PR TITLE
[ASV-1841] task: migrate to androidx

### PR DESCRIPTION
**Pull request by jdandrade**
**What does this PR do?**

   updates compile-sdk to 28 (min req to androidx). 
   removes BottomNavigationViewHelper class, disabling shifting now supported in view without reflection. 
   updates glide to 4.9.0. 
   updates parceler to 1.1.12.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] build.gradle

**How should this be manually tested?**

   Give a good run-through the app. I've tested InstalledIntentService because of the compile-sdk upgrade, just to be sure. target-sdk update would break it.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1841](https://aptoide.atlassian.net/browse/ASV-1841)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/proguard-rules.pro [hunk 1]
```diff
@@ -77,7 +77,7 @@
 -dontwarn com.adyen.ui.**
 
 ### disable shift animation bottom navigation ###
--keepclassmembers class android.support.design.internal.BottomNavigationMenuView {
+-keepclassmembers class com.google.android.material.bottomnavigation.BottomNavigationMenuView {
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/proguard-rules.pro]: True! nice catch

File: app/build.gradle [hunk 2]
```diff
@@ -358,73 +358,58 @@ dependencies {
...
-  //noinspection GradleDependency
-  implementation "com.android.support:recyclerview-v7:${SUPPORT_LIB_VERSION}"
-  //noinspection GradleDependency
-  implementation "com.android.support:support-v13:${SUPPORT_LIB_VERSION}"
-  //noinspection GradleDependency
-  implementation "com.android.support:design:${SUPPORT_LIB_VERSION}"
-  implementation "com.android.support:cardview-v7:${SUPPORT_LIB_VERSION}"
-  implementation "com.android.support:customtabs:${SUPPORT_LIB_VERSION}"
+
+  implementation "androidx.appcompat:appcompat:${APPCOMPAT_VERSION}"
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/build.gradle]: I would honestly prefer to wait until that version goes to production... We never know. But as soon as it releases... we bump it.

**Comment by jdandrade** [comment]
Merged and ready to be reviewed again.

**Comment by franciscoaleixo** [comment]
The changes done look good to me 👍

**Comment by jdandrade** [comment]
@phillipsgoncalves Please have a look into this & ad networks implications

**Comment by phillipsgoncalves** [comment]
Looks good to me

**Comment by jdandrade** [comment]
This branch commits were on v9.9.6.0, which is now merged to dev. Closed.

**Pull request closed without merging**